cores/arduino/printk.cpp

```diff
@@ -0,0 +1,230 @@
+					break;
+				case '2':
+					might_format |= 4;
+					goto still_format;
+					break;
+				case '4':
+					might_format |= 8;
+					goto still_format;
+					break;
+				case 'b':
+					binary_format = true;
+					goto still_format;
+					break;
+				case 'd':
+				case 'i':
+				case 'u':
+					if (!binary_format) {
+						unsigned long num =
+							va_arg(args,
+							       unsigned long);
+						unsigned long pos = 999999999;
+						unsigned long remainder = num;
+						int found_largest_digit = 0;
+
+						if (*fmt != 'u'
+						    && (num & (1 << 31))) {
+							if (len < size)
+								*dest++ =
+									'-',
+								len++;
+							num = (~num) + 1;
+							remainder = num;
+						}
+						while (pos >= 9) {
+							if (found_largest_digit
+							    || remainder >
+							    pos) {
+								found_largest_digit
+									= 1;
+								if (len < size)
+									*dest++
+										=
+											(
+												char)
+											((
+												 remainder
+												 / (
+													 pos
+													 +
+													 1))
+											 +
+											 48),
+									len++;
+								else
+									break;
+							}
+							remainder %= (pos + 1);
+							pos /= 10;
+						}
+						if (len < size)
+							*dest++ =
+								(char)(
+									remainder
+									+
+									48),
+							len++;
+						break;
+					}
+				case 'x':
+				case 'X':
+				case 'p': {
+					unsigned long num =
+						va_arg(args, unsigned long);
+					int sz = sizeof(num) * 2;
+
+					if (might_format & 8) {
+						sz = 4;
+					} else if (might_format & 4) {
+						sz = 2;
+					} else if (might_format & 2) {
+						sz = 1;
+					}
+					for (; sz; sz--) {
+						char nibble;
+						if (!binary_format) {
+							nibble =
+								(num >>
+								 ((sz -
+								   1) <<
+								  2) & 0xf);
+							nibble +=
+								nibble >
+								9 ? 87 : 48;
+						} else {
+							nibble =
+								(num >>
+								 ((sz -
+								   1) <<
+								  3) & 0xff);
+						}
+						if (len < size)
+							*dest++ =
+								nibble,
+							len++;
+						else
+							break;
+					}
+					break;
+				}
+				case 's': {
+					char *s = va_arg(args, char *);
+					while (*s)
+						if (len < size)
+							*dest++ =
+								*s++, len++;
+						else
+							break;
+					break;
+				}
+				case 'c': {
+					char c = va_arg(args, int);
+					if (len < size)
+						*dest++ = c, len++;
+					break;
+				}
+				default:
+					if (len < size)
+						*dest++ = '%', len++;
+					if (len < size)
+						*dest++ = *fmt, len++;
+					break;
+				}
+				might_format = 0;
+still_format:
+				(void)might_format;
+			}
+		}
+		++fmt;
+	}
+	*dest = '\0';
+	return len;
+}
+
+extern UARTClass Serial1;
+#define PRINTK_BUFSIZ 256
+void printk(const char *fmt, va_list args)
+{
+#ifdef CONFIGURE_DEBUG_CORELIB_ENABLED
+	int len = 0;
+
+    char tmp[PRINTK_BUFSIZ];
+
+    len = __vsnprintf(tmp, PRINTK_BUFSIZ, fmt, args);
+    
+    tmp[len] = '\0';
+    Serial1.println(tmp);
+#endif
+}
```

Comment:
I didn't change this file. It should be referenced from merge or rebase. I will fix this.